### PR TITLE
Add mouse reference genome GRCm38

### LINKED ISSUE #4137: add mouse genome to known references

### PR DESCRIPTION
**Pull request by tpoterba**
fixes #4137

### CODE REVIEW

File: src/main/scala/is/hail/variant/ReferenceGenome.scala [hunk 1]
```diff
@@ -527,8 +527,9 @@ case class ReferenceGenome(name: String, contigs: Array[String], lengths: Map[St
 
 object ReferenceGenome {
   var references: Map[String, ReferenceGenome] = Map()
-  val GRCh37: ReferenceGenome = fromResource("reference/grch37.json")
-  val GRCh38: ReferenceGenome = fromResource("reference/grch38.json")
+  val GRCh37: ReferenceGenome = fromResource("reference/GRCh37.json")
```
Comment (catoverdrive) [inline src/main/scala/is/hail/variant/ReferenceGenome.scala]: these should still be all lowercase names, right? Or did you also mean to change those file names?

Comment (tpoterba) [reply src/main/scala/is/hail/variant/ReferenceGenome.scala]: I did, I think they should be consistent with the reference genome names. can revert if you want

Comment (catoverdrive) [reply src/main/scala/is/hail/variant/ReferenceGenome.scala]: no, that looks good---I just don't see the renames in this PR.

Comment (tpoterba) [reply src/main/scala/is/hail/variant/ReferenceGenome.scala]: huh, looks like my file moves never got checked in. odd.

Comment (tpoterba) [reply src/main/scala/is/hail/variant/ReferenceGenome.scala]: will just revert

**Comment by tpoterba** [comment]
addressed comments! I haven't added a fasta / fai combination, but will make an issue to do that as a next step.

**Pull request merged**